test/service/test_manifest.py

```diff
@@ -788,3 +791,41 @@ def test_manifest_format_validation(self):
         url = self.base_url + '/manifest/files?format=invalid-type'
         response = requests.get(url)
         self.assertEqual(400, response.status_code, response.content)
+
+    @mock_sts
+    @mock_s3
+    def test_manifest_content_disposition_header(self):
+        self._index_canned_bundle(("f79257a7-dfc6-46d6-ae00-ba4b25313c10", "2018-09-14T133314.453337Z"))
+        for filters, expected_name, name_file in (
+                ({'project': {'is': ['Single of human pancreas']}}, 'Single of human pancreas ', True),
+                ({'project': {'is': ['Single of human pancreas', 'Mouse Melanoma']}}, 'hca-manifest-', False),
+                ({}, 'hca-manifest-', False)
+        ):
+            storage_service = StorageService()
+            storage_service.create_bucket()
+            for single_part in True, False:
+                with self.subTest(filters=filters, single_part=single_part):
+                    with mock.patch.object(type(config), 'disable_multipart_manifests', single_part):
+                        params = {
+                            'filters': json.dumps(filters),
+                            'format': 'full'
+                        }
+                        filters = AbstractService.parse_filters(params.get('filters'))
+                        estd = ElasticTransformDump()
+                        url = estd.transform_manifest(params.get('format', 'full'), filters).headers['Location']
+                        query = urlparse(url).query
+                        content_dispositions = parse_qs(query).get('response-content-disposition')
+                        if content_dispositions:
+                            content_disposition = one(content_dispositions)
+                            expected_prefix = 'attachment;filename=' + expected_name
+                            expected_suffix = '.tsv'
+                            actual_prefix = content_disposition[:len(expected_prefix)]
+                            actual_date = content_disposition[len(expected_prefix):-len(expected_suffix)]
+                            # Asserts if the date in the filename is correctly formatted
+                            if name_file:
+                                datetime.strptime(actual_date, "%Y-%m-%d %H.%M")
```

Comment:
Use mocks to pin the time so you can hard-code the expected file name. Whenever expectations are computed using logic from code under test, you risk duplicating a bug such that the bug in the expectation cancels out the bug in code under test.

https://github.com/DataBiosphere/azul/blob/ec89044d1ef8100749aa4459662d6d0f1969542e/test/service/test_manifest.py#L899